src/redux/sagas/extensions/extensionEnable.ts

```diff
@@ -100,20 +104,21 @@ function* extensionEnable({
         ),
       );
 
-      yield takeFrom(initialise.channel, ActionTypes.TRANSACTION_SUCCEEDED);
+      const result = yield waitForTxResult(initialise.channel);
 
       if (setUserRolesWithProofs) {
-        yield takeFrom(
-          setUserRolesWithProofs.channel,
-          ActionTypes.TRANSACTION_SUCCEEDED,
-        );
+        yield waitForTxResult(setUserRolesWithProofs.channel);
+        // assume if this doesn't error, the transaction has succeeded.
+        // @TODO: Handle state in which it gets cancelled
```

Comment:
Looks like right now if it gets cancelled the loading will go on forever. Is there a reason why we couldn't throw an error when the tx is cancelled? We could skip the extra if checks too

Reply:
@jakubcolony Yep, I think it makes sense to do this on the new ui branch as it's intrinsically connected to the cancel functionality in the ui